### ISSUE TITLE
Trouble with Latest Gradle plugin and LD Fatal Exception: java.lang.AbstractMethodError abstract method "java.util.Collection com.google.common.collect.Multimap.b(java.lang.Object)"

### ISSUE THREAD

**Issue by pranayairan** [launchdarkly/android-client#58]
Hello,

We are having trouble with latest launch darkly client and gradle plugin  classpath 'com.android.tools.build:gradle:3.3.0-alpha11'

After compiling the app and releasing it to stores, some of our users who are updating the app gets a crash, when we look at the crash reports we get following exception

```
Fatal Exception: java.lang.AbstractMethodError
abstract method "java.util.Collection com.google.common.collect.Multimap.b(java.lang.Object)"
```

With following stack trace : 

```
com.launchdarkly.android.UserLocalSharedPreferences.syncCurrentUserToActiveUser (UserLocalSharePreferences.java:26)
--
  | com.launchdarkly.android.UserManager.syncCurrentUserToActiveUserAndLog (UserManager.java:1)
  | com.launchdarkly.android.UserManager.saveFlagSettings (UserManager.java:7)
  | com.launchdarkly.android.UserManager.access$100 (UserManager.java:1)
  | com.launchdarkly.android.UserManager$2.onSuccess (UserManager.java:3)
  | com.launchdarkly.android.UserManager$2.onSuccess (UserManager.java:1)
  | com.google.common.util.concurrent.Futures$CallbackListener.run (Futures.java:2)
  | com.google.common.util.concurrent.SettableFuture.set (SettableFuture.java:1)
  | com.launchdarkly.android.HttpFeatureFlagFetcher$1.onResponse (HttpFeatureFlagFetcher.java:14)
  | okhttp3.RealCall$AsyncCall.execute (RealCall.java:4)
  | okhttp3.internal.NamedRunnable.run (NamedRunnable.java:3)
  | java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1167)
  | java.lang.Thread.run (Thread.java:764) 
```
We believe this is happening due to use of Guava android and some new changes in gradle plugin. This happens only to some users and there is no device or OS pattern. There is 1 common pattern of upgrade, ie when user trying to open and upgraded app compiled with 3.3.0-alpha11.

We were able to fix this issue by downgrading to 3.2.0, but even in that for some users the sync didn't happen properly and they still have a false value for feature flags that they are true for there id. 

The above error happens in ```LDClient.get().identify(ldUser)``` when we add identity of user to LD.

**Comment by pranayairan** [comment]
Also, one of our engineers dug deeper and thinks that 

when you are synchronizing the map, you are using internally, it requires calling a special API that you don't seem to call. 

See guava documentation on 
https://google.github.io/guava/releases/19.0/api/docs/com/google/common/collect/Multimaps.html#synchronizedMultimap(com.google.common.collect.Multimap)

**Comment by arun251** [comment]
Thanks for the report, we are taking a look.

**Comment by arun251** [comment]
Hi @pranayairan,

We couldn't reproduce the issue you were seeing. We don't advise releasing apps to production when building with alpha versions of build tools, but we did some refactoring that might resolve this issue for you. Let us know either way.

Thanks again for the report.

**Comment by arun251** [comment]
We believe this issue to be fixed in 2.5.4. Let us know if that's not the case.
